### PR TITLE
fix modal for edge provider

### PR DESCRIPTION
**Pull request by ahujsak**
Fixes Asana tasks 
Plugins Wallet Selector - Add ability that when user taps outside the window, the window will close.
https://app.asana.com/0/361770107085503/1128216423913281

MoonPay - Choose Your Wallet Screen - Tokens that are not supported by MoonPay will still appear in the list to choose from
https://app.asana.com/0/361770107085503/1128104739220755

MoonPay - Choose Your Wallet screen - Remove the list of coins shown ADA,BAT,BNB,NEO etc.
https://app.asana.com/0/361770107085503/1128104739220756

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

File: src/actions/CreateWalletActions.js [hunk 1]
```diff
@@ -54,6 +54,28 @@ export const createCurrencyWalletAndAddToSwap = (walletName: string, walletType:
...
+  try {
+    const wallet = await account.createCurrencyWallet(type, {
+      name: walletName,
+      fiatCurrencyCode,
+      keyOptions: format ? { format } : {}
+    })
+    return Promise.resolve(wallet)
+  } catch (e) {
+    return Promise.reject(e)
+  }
```
Comment (swansontec) [inline src/actions/CreateWalletActions.js]: Try catch is not needed, since that's the default anyhow.

File: src/actions/CreateWalletActions.js [hunk 2]
```diff
@@ -54,6 +54,28 @@ export const createCurrencyWalletAndAddToSwap = (walletName: string, walletType:
...
+  // return Promise.resolve('Hello')
```
Comment (swansontec) [inline src/actions/CreateWalletActions.js]: Um, hello back?

File: src/modules/UI/scenes/Plugins/EdgeProvider.js [hunk 3]
```diff
@@ -144,9 +163,35 @@ export class EdgeProvider extends Bridgeable {
...
+        return Promise.resolve(returnString)
+      } catch (e) {
+        const modal = createSimpleConfirmModal({
+          title: s.strings.create_wallet_failed_header,
+          message: s.strings.create_wallet_failed_message,
+          icon: <Icon type={MATERIAL_COMMUNITY} name={EXCLAMATION} size={30} />,
+          buttonText: s.strings.string_ok
+        })
+        launchModal(modal).then(response => {
+          Promise.reject(e)
```
Comment (swansontec) [inline src/modules/UI/scenes/Plugins/EdgeProvider.js]: No return.

File: src/actions/CreateWalletActions.js (hunk 1 above)
Comment (ahujsak) [reply src/actions/CreateWalletActions.js]: fixed

File: src/actions/CreateWalletActions.js (hunk 2 above)
Comment (ahujsak) [reply src/actions/CreateWalletActions.js]: fixed

File: src/modules/UI/scenes/Plugins/EdgeProvider.js (hunk 3 above)
Comment (ahujsak) [reply src/modules/UI/scenes/Plugins/EdgeProvider.js]: fixed

**Pull request merged**